src/utils/DownloadUtils.ts

```diff
@@ -21,4 +28,66 @@ export function addSolutionsToError(err: R2Error): void {
 export function generateProgressPercentage(progress: number, currentIndex: number, total: number): number {
     const completedProgress = (currentIndex / total) * 100;
     return completedProgress + (progress * 1/total);
-}
\ No newline at end of file
+}
+
+
+export async function getFullDependencyList(
+    profile: ImmutableProfile,
+    combos: ThunderstoreCombo[],
+): Promise<ThunderstoreCombo[]> {
+    const modList = await ProfileModList.getModList(profile);
```

Comment:
I don't think this has anything to do with the actual dependency resolution and shouldn't be here.

Reply:
I meant this shouldn't be done as part of the dependency resolution. Now it seems it was dropped altogether. I'm not sure why it's here, but I would wary of just dropping it.

Reply:
I made an archaelogical trip to the code and added this filtering stuff  back in. It is now in `getFullDependencyList` (in `DependencyUtils`).
I think there might be some sort of weird edge-case behaviour if you install multiple mods and modpacks with same mods (but different versions) as their dependencies, but I also think that was already the case before.

Reply:
the modlist is only used to filter out already installed mods, so I think getting the mod list in `getFullDependencyList` is ok. The list could also be passed as a parameter if you think that'd be better.